code/game/objects/obj_defense.dm

```diff
@@ -75,10 +75,11 @@
 /obj/bullet_act(obj/projectile/P)
 	. = ..()
 	playsound(src, P.hitsound, 50, TRUE)
-	if(P.suppressed != SUPPRESSED_VERY)
-		visible_message("<span class='danger'>[src] is hit by \a [P]!</span>", null, null, COMBAT_MESSAGE_RANGE)
-	if(!QDELETED(src)) //Bullet on_hit effect might have already destroyed this object
-		take_damage(P.damage, P.damage_type, P.flag, 0, turn(P.dir, 180), P.armour_penetration)
+	var/no_damage = TRUE
+	if(take_damage(P.damage, P.damage_type, P.flag, 0, turn(P.dir, 180), P.armour_penetration))
+		no_damage = FALSE
+	if(P.suppressed != SUPPRESSED_VERY && !QDELETED(src)) //Bullet on_hit effect might have already destroyed this object
+		visible_message("<span class='danger'>[src] is hit by \a [P][no_damage ? ", but it just bounces off" : ""]!</span>", null, null, COMBAT_MESSAGE_RANGE)
```

Comment:
Saying the projectiles "bounce off" stuff seems confusing considering projectiles actually can ricochet off objects, and it doesn't really work for things like lasers and disablers.

Reply:
That's a good point about beam weapons and actual ricochets. I'll word it some other way.